### ISSUE TITLE
Pacman -Fy fails after adding chaotic-aur

### ISSUE THREAD

**Issue by gardotd426** [PedroHLC/chaotic-aur#36]
So I already added chaotic-aur in Manjaro with no issues, but it seems something is wrong in vanilla arch. When I try to add it, add the key, and sign it, and then run `sudo pacman -Fy`, I get this:
```
ror: failed retrieving file 'chaotic-aur.files' from lonewolf-builder.duckdns.org : Maximum file size exceeded
error: failed retrieving file 'chaotic-aur.files' from chaotic.bangl.de : Maximum file size exceeded
error: failed retrieving file 'chaotic-aur.files' from repo.kitsuna.net : Maximum file size exceeded
error: failed to update chaotic-aur (download library error)
error: failed to synchronize all databases
```
Idk what's causing this, and I can't really figure anything out by searching, I followed the exact instructions from the chaotic-aur main page, lonewolf.pedrohlc.com

**Comment by PedroHLC** [comment]
Sorry about that, it was indeed something wrong in my end, not sure even when it happened.
That's something quite hard to fix, I'll have to stop all building services and recreate the database (for 1767 packages it won't be that fast).
It shouldn't happened in the first place, and I'll keep a look from now on.

I am going to close this issue only when I indeed fix it.

**Comment by gardotd426** [comment]
Hey man no need to apologize. That's a hell of a lot better of a response
than 90 percent of maintainers give when an issue like this gets posted. I
just wanted to make sure 1, it wasn't on my end, and 2, you were made aware
of the issue if it was indeed on your end. Thanks for the reply

**Comment by PedroHLC** [comment]
Ok, I was wrong, I recreated the database, and after a few very large packages, this happened again. This isn't my fault.
- The `chaotic-aur.files.tar.gz` is a total valid gziped tar.
- If you download the file manually, placing it in `/var/lib/pacman/sync` then I can run `pacman -Fl octopi`

It seems like it's time to file archlinux.org a bug report :sweat_smile:

**Comment by PedroHLC** [comment]
Hi @gardotd426, any plan on your side on submitting this issue to archlinux?

**Comment by gardotd426** [comment]
Whoa, I'm sorry I somehow missed your last message, no idea how that
happened. Let me try adding it to Arch again, and if it still doesn't work
we'll go the bug report route. Because like I said, it added to Manjaro
with zero issues. Let me test it out I'll get right back to you.

**Comment by gardotd426** [comment]
Yep. Same issue:
```
sudo pacman -Fy
:: Synchronizing package databases...
 core                                                                   846.8 KiB   492 KiB/s 00:02 [###########################################################] 100%
 extra                                                                    8.9 MiB   601 KiB/s 00:15 [###########################################################] 100%
 community                                                               19.8 MiB   522 KiB/s 00:39 [###########################################################] 100%
 multilib                                                               243.2 KiB   816 KiB/s 00:00 [###########################################################] 100%
 valveaur                                                               204.6 KiB   731 KiB/s 00:00 [###########################################################] 100%
 valveaur.sig                                                           566.0   B  0.00   B/s 00:00 [###########################################################] 100%
error: failed retrieving file 'chaotic-aur.files' from lonewolf-builder.duckdns.org : Maximum file size exceeded
error: failed retrieving file 'chaotic-aur.files' from chaotic.bangl.de : Maximum file size exceeded
error: failed retrieving file 'chaotic-aur.files' from chaotic.bangl.de : Maximum file size exceeded
error: failed retrieving file 'chaotic-aur.files' from repo.kitsuna.net : Maximum file size exceeded
error: failed to update chaotic-aur (download library error)
error: failed to synchronize all databases
```
I've never filed a bug report with Arch, I've only filed actual bug reports (rather than git issues) with KDE and they automate everything, so if there's anything you know that I should include to help me out, or if you want to help me out with filing then that would be awesome. So when you say it happened again, you're saying that you get the same issue? As in, you have to place the tar into /var/lib/pacman/sync in order for it to pass?

**Comment by gardotd426** [comment]
Also, where can I get `chaotic-aur.files.tar.gz`? I'm having trouble finding it over on lonewolf.pedrohlc.com

**Comment by PedroHLC** [comment]
Yes, I get the same issue. And yes, if you download and copy manually, the other's `-F*` commands will work without problems.
http://lonewolf-builder.duckdns.org/chaotic-aur/x86_64/chaotic-aur.files

**Comment by gardotd426** [comment]
Yeah I placed `chaotic-aur.files` into `/var/lib/pacman/sync` and I didn't
get the error with `pacman -Fl` or `pacman -Fy`. I'm researching now how
exactly Arch wants bug reports filed, because I know how particular the
community is and don't wanna annoy them lol, but I'll give you the link
when I file if you wanna add to it or comment or whatever. This seems
pretty straightforward though, I mean it's reproducible, chaotic-aur is
listed on the arch wiki as a signed community repo (it's not like it's some
obscure repo no one knows will work or not), it works on Manjaro, so it
works downstream, pacman is directly under Arch's responsibility, so I
mean, pretty clearly seems like it's a legit bug.

**Comment by PedroHLC** [comment]
I agree, and it's their tool (`repo-add`) which generated the `.files` in the first place.
I'll keep an eye. Thanks for the contribution! :smile:

**Comment by gardotd426** [comment]
[Here you go](https://bugs.archlinux.org/task/65197)

**Comment by gardotd426** [comment]
Seems like we're not going to get anywhere. Apparently you've just gone over the limit of number of packages for your repo, Robin from the bugtracker page says: "There is a hard limit on database sizes that pacman enforces." However, that makes little sense to me (admittedly I know nothing about this), but that would mean that chaotic-aur would HAVE to be larger than the extra or community official arch repos, which it's not from what I can see. I'm gonna need your help over there because I don't know enough about this to rebut that comment other than what I just said.

**Comment by gardotd426** [comment]
Nope, nevermind. Apparently Manjaro does patch their version of pacman to raise the database filesize limit, vanilla Arch places a limit of 25MB on pacman repo database files, and the chaotic-aur.files database file is 30MB, and I doubt they're going to change pacman in order to accommodate a community repository, but I don't know. But yeah, it's not an actual bug, from what it seems.

**Comment by gardotd426** [comment]
[Here](https://gitlab.manjaro.org/packages/core/pacman/commit/cc8253a7bd6c230c533db448cc1daad66e96d498) is the link for the manjaro pacman patch that shows that they indeed raise the 25MB limit because their own community repo is larger than that, so they patch it to make the max size 32MB. So if the `chaotic-aur` repo files list file was 3MB larger, it wouldn't work on Manjaro either, it seems. I'm not sure what could be done to fix this aside from shrinking the repo 5MB, y'know?

**Comment by gardotd426** [comment]
Yeah, they closed it. Not a bug.

**Comment by gardotd426** [comment]
Welp... Nevermind. Some angel came by and reopened it, and said "why was this closed as not a bug? Clearly pacman needs adjusted here." And then the original guy that closed it suggested maybe changing pacman to have the limit adjusted in `etc/pacman.conf` instead of at compile-time. Now it's been edited by someone on the team to say that it's due in version 5.2.2, so it seems they might actually fix it in pacman itself yayyyyyyy

**Comment by gardotd426** [comment]
Alan submitted a patch bumping the limit to 128MB for now, and they'll talk about letting it be a config option in the future. But yeah, whenever it hits it should be good.

**Comment by PedroHLC** [comment]
I see, I can try to replace the compression algorithm in the mean time.
Thanks for your efforts!

**Comment by gardotd426** [comment]
No problem. I try to contribute as much as I can, we're all in this
together. Anyway yeah, on the bug tracker page someone said using zstd
instead of gz would help (they said you could shrink the size by around
25%, not sure if that's true or not)

**Comment by gardotd426** [comment]
Hey so I decompressed `chaotic-aur.files`, and tried recompressing it with `ZSTD_CLEVEL=19 tar --zstd -cf ...` to see what happens, and if you use zstd instead of gzip it actually does make it 19MB. And I then decompressed it again to make sure that it actually worked, and it does.

**Comment by PedroHLC** [comment]
I'm unable to set the compression level to `repo-add` (at least easily). But I liked how fast it's with this preset.

**Comment by gardotd426** [comment]
I don't know how repo-add does it, I just know that `tar` uses the
`ZSTD_CLEVEL=` envvar. Now I'm trying to get the mirrors to speed up
because I can't crack 90 kb/s with chaotic even though I get 1.3MB/s on all
my actual repos. I'm gonna try just using the American mirror and see if
that  helps. But anyway have you tried the envvar with repo-add?

**Comment by gardotd426** [comment]
I don't know how repo-add does it, I just know that `tar` uses the
`ZSTD_CLEVEL=` envvar. Now I'm trying to get the mirrors to speed up
because I can't crack 90 kb/s with chaotic even though I get 1.3MB/s on all
my actual repos. I'm gonna try just using the American mirror and see if
that  helps. But anyway have you tried the envvar with repo-add?

**Comment by gardotd426** [comment]
Hey just fyi Eli over at Arch just told me "repo-add would respect the `COMPRESSZST=()` tuneable once I polish up https://patchwork.archlinux.org/patch/1042/"

**Comment by PedroHLC** [comment]
That's wonderful news

**Comment by gardotd426** [comment]
Yeah, and they're discussing which way to go on raising or allowing
user-configuration of the database files size limits, and it seems one way
or another it'll get fixed, but I definitely suggest you come chime in over
on the thread if you get time. I'm not nearly as knowledgeable on this
stuff, so I'm basically over there like "one way or another, as long as
y'all make it work" lol.

**Comment by PedroHLC** [comment]
Working well with the latest pacman and database.
Let's hope they conclude their enhancement thought.
Thanks @gardotd426. I'll be closing this as it's not directly affecting us anymore.

**Comment by PedroHLC** [comment]
We got too big again!

**Comment by gardotd426** [comment]
Eh godddd. Even with zstd? 

And honestly, I hate to say it, because I use almost every single one of his packages (customized and built myself mostly, rather than prebuilts from chaotic, but still), but it might be time to consider trimming some of the huge number of variants of Etienne's packages. I mean, you host maybe 15 actually different packages, but each one has like 10 variations to the point where you literally have 196 TKG packages. With like 30 versions of tkg wine, tkg kernels, etc. That's going to be hard to keep under limit. 

And when I filed the bug report with the Arch devs, you know how they are, they were VERY averse to raising any limits, and only barely raised it, I don't think they're going to raise it again, they're very strict and it was so recently they had to raise it the first time that they'll feel like we'll want them to raise it every couple months, when it had been years and years before they'd had to adjust it until we came along, haha. I think it had been over a decade, I don't remember exactly but I know one of them mentioned it in the bug report thread. 

So that would seem to leave only either better compression, or trimming of some packages. Do you have any ideas? 

I mean the tkg kernel variants are probably the biggest takers of space here, but maybe also prebuilts of AUR packages, maybe there are some barely-ever-used ones you can just leave to the AUR? 

And just for example, here are *just* the tkg builds you've got currently:
```
chaotic-aur chaotic-nvidia-dev-dkms-tkg 440.66.14-1
chaotic-aur chaotic-nvidia-dev-egl-wayland-tkg 440.66.14-1
chaotic-aur chaotic-nvidia-dev-opencl-tkg 440.66.14-1
chaotic-aur chaotic-nvidia-dev-utils-tkg 440.66.14-1
chaotic-aur chaotic-nvidia-dkms-tkg 440.82-200
chaotic-aur chaotic-nvidia-egl-wayland-tkg 440.82-200
chaotic-aur chaotic-nvidia-opencl-tkg 440.82-200
chaotic-aur chaotic-nvidia-utils-tkg 440.82-200
chaotic-aur lib32-chaotic-nvidia-dev-opencl-tkg 440.66.14-1
chaotic-aur lib32-chaotic-nvidia-dev-utils-tkg 440.66.14-1
chaotic-aur lib32-chaotic-nvidia-opencl-tkg 440.82-200
chaotic-aur lib32-chaotic-nvidia-utils-tkg 440.82-200
chaotic-aur lib32-mesa-tkg-git 20.2.0_devel.123587.f7fcbe9830d-1
chaotic-aur lib32-spirv-tools-tkg-git 2020.2.r47.gc8590c18-1
chaotic-aur linux-lts-tkg-bmq 5.4.40-56
chaotic-aur linux-lts-tkg-bmq-headers 5.4.40-56
chaotic-aur linux-lts-tkg-muqss 5.4.40-56
chaotic-aur linux-lts-tkg-muqss-headers 5.4.40-56
chaotic-aur linux-lts-tkg-pds 5.4.40-56
chaotic-aur linux-lts-tkg-pds-headers 5.4.40-56
chaotic-aur linux-tkg-bmq 5.6.12-24
chaotic-aur linux-tkg-bmq-atom 5.6.12-24
chaotic-aur linux-tkg-bmq-atom-headers 5.6.12-24
chaotic-aur linux-tkg-bmq-barcelona 5.6.12-24
chaotic-aur linux-tkg-bmq-barcelona-headers 5.6.12-24
chaotic-aur linux-tkg-bmq-bobcat 5.6.12-24
chaotic-aur linux-tkg-bmq-bobcat-headers 5.6.12-24
chaotic-aur linux-tkg-bmq-broadwell 5.6.12-24
chaotic-aur linux-tkg-bmq-broadwell-headers 5.6.12-24
chaotic-aur linux-tkg-bmq-bulldozer 5.6.12-24
chaotic-aur linux-tkg-bmq-bulldozer-headers 5.6.12-24
chaotic-aur linux-tkg-bmq-cannonlake 5.6.12-25
chaotic-aur linux-tkg-bmq-cannonlake-headers 5.6.12-25
chaotic-aur linux-tkg-bmq-core2 5.6.12-25
chaotic-aur linux-tkg-bmq-core2-headers 5.6.12-25
chaotic-aur linux-tkg-bmq-excavator 5.6.12-25
chaotic-aur linux-tkg-bmq-excavator-headers 5.6.12-25
chaotic-aur linux-tkg-bmq-haswell 5.6.12-25
chaotic-aur linux-tkg-bmq-haswell-headers 5.6.12-25
chaotic-aur linux-tkg-bmq-headers 5.6.12-24
chaotic-aur linux-tkg-bmq-icelake 5.6.12-25
chaotic-aur linux-tkg-bmq-icelake-headers 5.6.12-25
chaotic-aur linux-tkg-bmq-ivybridge 5.6.12-25
chaotic-aur linux-tkg-bmq-ivybridge-headers 5.6.12-25
chaotic-aur linux-tkg-bmq-jaguar 5.6.12-25
chaotic-aur linux-tkg-bmq-jaguar-headers 5.6.12-25
chaotic-aur linux-tkg-bmq-k10 5.6.12-25
chaotic-aur linux-tkg-bmq-k10-headers 5.6.12-25
chaotic-aur linux-tkg-bmq-k8 5.6.12-25
chaotic-aur linux-tkg-bmq-k8-headers 5.6.12-25
chaotic-aur linux-tkg-bmq-k8sse3 5.6.12-25
chaotic-aur linux-tkg-bmq-k8sse3-headers 5.6.12-25
chaotic-aur linux-tkg-bmq-mpsc 5.6.12-25
chaotic-aur linux-tkg-bmq-mpsc-headers 5.6.12-25
chaotic-aur linux-tkg-bmq-nehalem 5.6.12-25
chaotic-aur linux-tkg-bmq-nehalem-headers 5.6.12-25
chaotic-aur linux-tkg-bmq-piledriver 5.6.12-25
chaotic-aur linux-tkg-bmq-piledriver-headers 5.6.12-25
chaotic-aur linux-tkg-bmq-sandybridge 5.6.12-25
chaotic-aur linux-tkg-bmq-sandybridge-headers 5.6.12-25
chaotic-aur linux-tkg-bmq-silvermont 5.6.12-25
chaotic-aur linux-tkg-bmq-silvermont-headers 5.6.12-25
chaotic-aur linux-tkg-bmq-skylake 5.6.12-25
chaotic-aur linux-tkg-bmq-skylake-headers 5.6.12-25
chaotic-aur linux-tkg-bmq-skylakex 5.6.12-25
chaotic-aur linux-tkg-bmq-skylakex-headers 5.6.12-25
chaotic-aur linux-tkg-bmq-steamroller 5.6.12-25
chaotic-aur linux-tkg-bmq-steamroller-headers 5.6.12-25
chaotic-aur linux-tkg-bmq-westmere 5.6.12-25
chaotic-aur linux-tkg-bmq-westmere-headers 5.6.12-25
chaotic-aur linux-tkg-bmq-zen 5.6.12-25
chaotic-aur linux-tkg-bmq-zen-headers 5.6.12-25
chaotic-aur linux-tkg-bmq-zen2 5.6.12-25
chaotic-aur linux-tkg-bmq-zen2-headers 5.6.12-25
chaotic-aur linux-tkg-muqss 5.6.12-25
chaotic-aur linux-tkg-muqss-atom 5.6.12-25
chaotic-aur linux-tkg-muqss-atom-headers 5.6.12-25
chaotic-aur linux-tkg-muqss-barcelona 5.6.12-25
chaotic-aur linux-tkg-muqss-barcelona-headers 5.6.12-25
chaotic-aur linux-tkg-muqss-bobcat 5.6.12-25
chaotic-aur linux-tkg-muqss-bobcat-headers 5.6.12-25
chaotic-aur linux-tkg-muqss-broadwell 5.6.12-25
chaotic-aur linux-tkg-muqss-broadwell-headers 5.6.12-25
chaotic-aur linux-tkg-muqss-bulldozer 5.6.12-26
chaotic-aur linux-tkg-muqss-bulldozer-headers 5.6.12-26
chaotic-aur linux-tkg-muqss-cannonlake 5.6.12-27
chaotic-aur linux-tkg-muqss-cannonlake-headers 5.6.12-27
chaotic-aur linux-tkg-muqss-core2 5.6.12-27
chaotic-aur linux-tkg-muqss-core2-headers 5.6.12-27
chaotic-aur linux-tkg-muqss-excavator 5.6.12-27
chaotic-aur linux-tkg-muqss-excavator-headers 5.6.12-27
chaotic-aur linux-tkg-muqss-haswell 5.6.12-27
chaotic-aur linux-tkg-muqss-haswell-headers 5.6.12-27
chaotic-aur linux-tkg-muqss-headers 5.6.12-25
chaotic-aur linux-tkg-muqss-icelake 5.6.12-27
chaotic-aur linux-tkg-muqss-icelake-headers 5.6.12-27
chaotic-aur linux-tkg-muqss-ivybridge 5.6.12-27
chaotic-aur linux-tkg-muqss-ivybridge-headers 5.6.12-27
chaotic-aur linux-tkg-muqss-jaguar 5.6.12-27
chaotic-aur linux-tkg-muqss-jaguar-headers 5.6.12-27
chaotic-aur linux-tkg-muqss-k10 5.6.12-27
chaotic-aur linux-tkg-muqss-k10-headers 5.6.12-27
chaotic-aur linux-tkg-muqss-k8 5.6.12-27
chaotic-aur linux-tkg-muqss-k8-headers 5.6.12-27
chaotic-aur linux-tkg-muqss-k8sse3 5.6.12-27
chaotic-aur linux-tkg-muqss-k8sse3-headers 5.6.12-27
chaotic-aur linux-tkg-muqss-mpsc 5.6.12-27
chaotic-aur linux-tkg-muqss-mpsc-headers 5.6.12-27
chaotic-aur linux-tkg-muqss-nehalem 5.6.12-27
chaotic-aur linux-tkg-muqss-nehalem-headers 5.6.12-27
chaotic-aur linux-tkg-muqss-piledriver 5.6.12-27
chaotic-aur linux-tkg-muqss-piledriver-headers 5.6.12-27
chaotic-aur linux-tkg-muqss-sandybridge 5.6.12-27
chaotic-aur linux-tkg-muqss-sandybridge-headers 5.6.12-27
chaotic-aur linux-tkg-muqss-silvermont 5.6.12-27
chaotic-aur linux-tkg-muqss-silvermont-headers 5.6.12-27
chaotic-aur linux-tkg-muqss-skylake 5.6.12-27
chaotic-aur linux-tkg-muqss-skylake-headers 5.6.12-27
chaotic-aur linux-tkg-muqss-skylakex 5.6.12-27
chaotic-aur linux-tkg-muqss-skylakex-headers 5.6.12-27
chaotic-aur linux-tkg-muqss-steamroller 5.6.12-27
chaotic-aur linux-tkg-muqss-steamroller-headers 5.6.12-27
chaotic-aur linux-tkg-muqss-westmere 5.6.12-27
chaotic-aur linux-tkg-muqss-westmere-headers 5.6.12-27
chaotic-aur linux-tkg-muqss-zen 5.6.12-27
chaotic-aur linux-tkg-muqss-zen-headers 5.6.12-27
chaotic-aur linux-tkg-muqss-zen2 5.6.12-27
chaotic-aur linux-tkg-muqss-zen2-headers 5.6.12-27
chaotic-aur linux-tkg-pds 5.6.12-27
chaotic-aur linux-tkg-pds-atom 5.6.12-27
chaotic-aur linux-tkg-pds-atom-headers 5.6.12-27
chaotic-aur linux-tkg-pds-barcelona 5.6.12-27
chaotic-aur linux-tkg-pds-barcelona-headers 5.6.12-27
chaotic-aur linux-tkg-pds-bobcat 5.6.12-27
chaotic-aur linux-tkg-pds-bobcat-headers 5.6.12-27
chaotic-aur linux-tkg-pds-broadwell 5.6.12-27
chaotic-aur linux-tkg-pds-broadwell-headers 5.6.12-27
chaotic-aur linux-tkg-pds-bulldozer 5.6.12-27
chaotic-aur linux-tkg-pds-bulldozer-headers 5.6.12-27
chaotic-aur linux-tkg-pds-cannonlake 5.6.12-27
chaotic-aur linux-tkg-pds-cannonlake-headers 5.6.12-27
chaotic-aur linux-tkg-pds-core2 5.6.12-27
chaotic-aur linux-tkg-pds-core2-headers 5.6.12-27
chaotic-aur linux-tkg-pds-excavator 5.6.12-27
chaotic-aur linux-tkg-pds-excavator-headers 5.6.12-27
chaotic-aur linux-tkg-pds-haswell 5.6.12-27
chaotic-aur linux-tkg-pds-haswell-headers 5.6.12-27
chaotic-aur linux-tkg-pds-headers 5.6.12-27
chaotic-aur linux-tkg-pds-icelake 5.6.12-27
chaotic-aur linux-tkg-pds-icelake-headers 5.6.12-27
chaotic-aur linux-tkg-pds-ivybridge 5.6.12-27
chaotic-aur linux-tkg-pds-ivybridge-headers 5.6.12-27
chaotic-aur linux-tkg-pds-jaguar 5.6.12-27
chaotic-aur linux-tkg-pds-jaguar-headers 5.6.12-27
chaotic-aur linux-tkg-pds-k10 5.6.11-23
chaotic-aur linux-tkg-pds-k10-headers 5.6.11-23
chaotic-aur linux-tkg-pds-k8 5.6.11-23
chaotic-aur linux-tkg-pds-k8-headers 5.6.11-23
chaotic-aur linux-tkg-pds-k8sse3 5.6.11-23
chaotic-aur linux-tkg-pds-k8sse3-headers 5.6.11-23
chaotic-aur linux-tkg-pds-mpsc 5.6.11-23
chaotic-aur linux-tkg-pds-mpsc-headers 5.6.11-23
chaotic-aur linux-tkg-pds-nehalem 5.6.11-23
chaotic-aur linux-tkg-pds-nehalem-headers 5.6.11-23
chaotic-aur linux-tkg-pds-piledriver 5.6.11-23
chaotic-aur linux-tkg-pds-piledriver-headers 5.6.11-23
chaotic-aur linux-tkg-pds-sandybridge 5.6.11-23
chaotic-aur linux-tkg-pds-sandybridge-headers 5.6.11-23
chaotic-aur linux-tkg-pds-silvermont 5.6.11-23
chaotic-aur linux-tkg-pds-silvermont-headers 5.6.11-23
chaotic-aur linux-tkg-pds-skylake 5.6.11-23
chaotic-aur linux-tkg-pds-skylake-headers 5.6.11-23
chaotic-aur linux-tkg-pds-skylakex 5.6.12-24
chaotic-aur linux-tkg-pds-skylakex-headers 5.6.12-24
chaotic-aur linux-tkg-pds-steamroller 5.6.12-24
chaotic-aur linux-tkg-pds-steamroller-headers 5.6.12-24
chaotic-aur linux-tkg-pds-westmere 5.6.12-24
chaotic-aur linux-tkg-pds-westmere-headers 5.6.12-24
chaotic-aur linux-tkg-pds-zen 5.6.12-24
chaotic-aur linux-tkg-pds-zen-headers 5.6.12-24
chaotic-aur linux-tkg-pds-zen2 5.6.12-24
chaotic-aur linux-tkg-pds-zen2-headers 5.6.12-24
chaotic-aur mesa-tkg-git 20.2.0_devel.123587.f7fcbe9830d-1
chaotic-aur proton-tkg-git 5.8.r1.g3e366129-1
chaotic-aur spirv-tools-tkg-git 2020.2.r47.gc8590c18-1
chaotic-aur vulkan-headers-tkg-git 1.2.140.r0.g9250d5a-1 [installed]
chaotic-aur wine-tkg-fsync-git 5.8.r34.gb071d37b80-270
chaotic-aur wine-tkg-fsync-opt-git 5.8.r34.gb071d37b80-270
chaotic-aur wine-tkg-fsync-vkd3d-git 5.8.r34.gb071d37b80-270
chaotic-aur wine-tkg-fsync-vkd3d-opt-git 5.8.r34.gb071d37b80-270
chaotic-aur wine-tkg-git-leagueoflegends 5.8.r0.g7d92cd31-270
chaotic-aur wine-tkg-git-leagueoflegends-opt 5.8.r0.g7d92cd31-270
chaotic-aur wine-tkg-staging-fsync-git 5.8.r1.g3e366129-270
chaotic-aur wine-tkg-staging-fsync-opt-git 5.8.r1.g3e366129-270
chaotic-aur wine-tkg-staging-fsync-vkd3d-git 5.8.r1.g3e366129-270
chaotic-aur wine-tkg-staging-fsync-vkd3d-opt-git 5.8.r1.g3e366129-270
```

There's like 100 different kernel variations. I definitely think you could maybe consider dropping some stuff like, linux-tkg-pds, bmq and muqss -westmere, -steamroller, -jaguar, -silvermont, -bobcat, -core2, -atom, etc. Having a pds, bmq, and muqss kernel build for literally every CPU architecture in the last like 15 years is far too much for one repository, y'know?

**Comment by gardotd426** [comment]
Wait, I thought the limit used to be 25, and they had bumped the limit to 128? Your db is only 27MB right now, nowhere near the 128. Maybe they decreased it again, but they were supposed to add a config option in pacman.conf for the users to be able to raise the limit before they lowered it again. 

Either they screwed us and lowered the limit back to 25 without adding the config option, or something else is going on

**Comment by gardotd426** [comment]
Also, another development, it took me FOREVER to find, because in my archlinux bug tracker account overview it showed no filed reports, but then when I finally found it, it says I was the one that reported it, so there's another bug, in their website lol, but anyway...

Right now it's still considered reopened, I don't know if it was closed again back after everything was temporarily settled and then recently was reopened, or if it never got closed, but I posted explaining that you're back to being 2MB over the limit again, and asking if the 128MB limit they temporarily put in place had been revoked for some reason, and if so, if maybe they could raise it back up a little bit, even if not to 128MB. 

Bug tracker thread:

https://bugs.archlinux.org/task/65197

**Comment by gardotd426** [comment]
Jesus Christ. We're boned, apparently. Eli Schwartz says "The limit in git is 128MB, but it hasn't been backported to the current release series." 

I'm really confused, so did they never raise the limit to 128MB? Did you just cut corners to get under the 25? Or did you use the zstd to get under 25, and we never had the benefit of 128?

Either way, they're not fixing it until the next release of pacman, at which point it will indeed be 128.

**Comment by PedroHLC** [comment]
Man, you're fast!

> Did you just cut corners to get under the 25?
I think zstd may have done it.

> Either way, they're not fixing it until the next release of pacman, at which point it will indeed be 128.
That's a solution, of sorts :sweat_smile: 

> Having a pds, bmq, and muqss kernel build for literally every CPU architecture in the last like 15 years is far too much for one repository, y'know?
Yeah, I need to refactor chaotic for the usage inside the new clouds, which uses Kubernets, I may move TkG, Librewish, and others non-AUR packages to "sub-repos".

Thanks for the fast insight!

**Comment by gardotd426** [comment]
> Yeah, I need to refactor chaotic for the usage inside the new cloud, which uses Kubernets, I may move TkG, Librewish, and others non-AUR packages to "sub-repos".

Hey don't do that! Lol but seriously, what I would suggest is keeping all the tkg packages that are the ones you don't customize at all, like vulkan drivers, spirv, mingw, etc, because there aren't a million variants, and there's not really any customization needed, and everyone can still easily get prebuilt packages for them from you, which is what I do and I imagine most other people do. But for the kernels, wine and proton, I run his PKGBUILDs myself to customize all the fantastic options. 

But like, you only have ONE proton-tkg, but like a million wine-tkgs, even though both can be equally customized in the config options, and there's no real reason to have a million of one and only one of the other, y'know? So if you're good with only one proton, maybe drop a few of the wines, and just offer a couple main configs, like wine-tkg-protonified-fsync-faudio, because that's what everyone's going to want, otherwise they might as well just use wine-staging. 

The biggest thing is the kernels, all the million variants for old-ass architectures no one is using. I would almost guarantee that a good number of them have never been installed by anyone. Sure, keep the zen, zen2, skylake, etc. builds, but I'd get rid of the 10-year-old CPU builds.

I mean obviously it's up to you, but I feel invested at this point lmao.

**Comment by gardotd426** [comment]
Hey man! The Arch guys closed the bug and cited a commit, I went over to the pacman source repo, and 5.2.2 is now released in Arch Linux! It's not in the repos yet, but I went ahead and did an `asp checkout` and edited the PKGBUILD to pull in the new version, and sure enough, everything works! The new limit is 128MB. It should just be a matter of time, 5.2.2's release source tarball went up at the source repo 7 days ago, it might already be in the testing repo.

**Comment by gardotd426** [comment]
I'll go ahead and close this once it goes live, or you can.

**Comment by PedroHLC** [comment]
Hey, thanks for the update!!!!
